src/scripts/main.js

```diff
@@ -1,5 +1,31 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  const wallRect = wall.getBoundingClientRect();
+
+  if (
+    e.clientX < wallRect.left ||
+    e.clientX > wallRect.right ||
+    e.clientY < wallRect.top ||
+    e.clientY > wallRect.bottom
+  ) {
+    return;
+  }
+
+  const spiderRect = spider.getBoundingClientRect();
```

Comment:
Similar to the previous comment, ensure that the 'spider' element exists before calling `getBoundingClientRect()`, as this will throw an error if `spider` is `null`.